src/Tools/ExternalAccess/Razor/Features/Microsoft.CodeAnalysis.ExternalAccess.Razor.Features.csproj

```diff
@@ -21,6 +21,11 @@
   </ItemGroup>
 
   <ItemGroup>
+    <!--
+      This is used to load in the language server so it gets special treatment. Otherwise only razor assemblies should be added.
+    -->
+    <InternalsVisibleTo Include="Microsoft.CodeAnalysis.LanguageServer" />
```

Comment:
Isn't this going to cause a problem later, when we need the language server to have IVT to this project, so we can put LSP endpoints in it? Or is that going to go through the `IRazorServiceProvider`?

Reply:
Might need to chat with you. My understanding of how the lsp endpoints got hooked up is just a normal export with a razor class in the middle. That's what I thought co-hosting was doing.

Can we hook up endpoints through the service provider? Might make things easier and cleaner

Reply:
It's an export, yeah, but the type being exported has to inherit from something like `ILspServiceRequestHandler` which we expose via our EA.

BUT I just looked it up to be sure, and actually I'm wrong, and I think this will be fine, because that type is in MS.CA.LS.Protocol, so maybe we can get away with never needing to expose anything from the language server itself through our EA.